mmcls/models/backbones/gcvit.py

```diff
@@ -0,0 +1,792 @@
+    Transformers <https://arxiv.org/abs/2206.09959>"."""
+
+    def __init__(self,
+                 dim,
+                 depth,
+                 input_resolution,
+                 num_heads,
+                 window_size,
+                 downsample=True,
+                 mlp_ratio=4.,
+                 qkv_bias=True,
+                 qk_scale=None,
+                 drop=0.,
+                 attn_drop=0.,
+                 drop_path=0.,
+                 norm_layer=nn.LayerNorm,
+                 layer_scale=None):
+        """
+        Args:
+            dim: feature size dimension.
+            depth: number of layers in each stage.
+            input_resolution: input image resolution.
+            window_size: window size in each stage.
+            downsample: bool argument for down-sampling.
+            mlp_ratio: MLP ratio.
+            num_heads: number of heads in each stage.
+            qkv_bias: bool argument for query, key, value learnable bias.
+            qk_scale: bool argument to scaling query, key.
+            drop: dropout rate.
+            attn_drop: attention dropout rate.
+            drop_path: drop path rate.
+            norm_layer: normalization layer.
+            layer_scale: layer scaling coefficient.
+        """
+
+        super().__init__()
+        self.blocks = nn.ModuleList([
+            GCViTBlock(
+                dim=dim,
+                num_heads=num_heads,
+                window_size=window_size,
+                mlp_ratio=mlp_ratio,
+                qkv_bias=qkv_bias,
+                qk_scale=qk_scale,
+                attention=WindowAttention if
+                (i % 2 == 0) else WindowAttentionGlobal,
+                drop=drop,
+                attn_drop=attn_drop,
+                drop_path=drop_path[i]
+                if isinstance(drop_path, list) else drop_path,
+                norm_layer=norm_layer,
+                layer_scale=layer_scale,
+                input_resolution=input_resolution) for i in range(depth)
+        ])
+        self.downsample = None if not downsample else ReduceSize(
+            dim=dim, norm_layer=norm_layer)
+        self.q_global_gen = GlobalQueryGen(dim, input_resolution, window_size,
+                                           num_heads)
+
+    def forward(self, x):
+        q_global = self.q_global_gen(_to_channel_first(x))
+        for blk in self.blocks:
+            x = blk(x, q_global)
+        if self.downsample is None:
+            return x
+        return self.downsample(x)
+
+
+@MODELS.register_module()
+class GCViT(nn.Module):
+    """GCViT based on: "Hatamizadeh et al., Global Context Vision Transformers.
+
+    <https://arxiv.org/abs/2206.09959>".
+    """
+    arch_zoo = {
+        **dict.fromkeys(['xxt', 'xxtiny'],
+                        {'depths': [2, 2, 6, 2],
+                         'num_heads': [2, 4, 8, 16],
+                         'window_size': [7, 7, 14, 7],
+                         'dim': 64,
```

Comment:
the parameters with the same value should be remove from this dict, and put them in `__init__` function.